### ISSUE TITLE
Missmatching types between languageserver and vscode

### ISSUE THREAD

**Issue by timrc** [microsoft/vscode-languageserver-node#1263]
I might be completely in error here, but it seems as there are quite a few types miss-matching happening between vscode and languageserver.

To dive into this, I'll take the TypeHierarchyItem as an example.

In VS Code ([vscode.d.ts](https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts#L5462)), this has a field called uri which is of type Uri - and it's defined [here](https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts#L1315) and it has been like that in older version of VS Code types too.

In language server, the same type, also has a field [uri](https://github.com/microsoft/vscode-languageserver-node/blob/release/types/3.17.3/types/src/main.ts#L3724), but this one is of type DocumentUri, which is simply [defined as a string](https://github.com/microsoft/vscode-languageserver-node/blob/release/types/3.17.3/types/src/main.ts#L10)
It's the same in older versions too: https://github.com/microsoft/vscode-languageserver-node/blob/release/types/3.17.0/types/src/main.ts#L3713 

To dive even deeper, there's an interface for this in VS Code, https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/typeHierarchy/common/typeHierarchy.ts#L32, which uri type is URI.

Given the same item TypeHierarchyItem, its field uri, is defined in 3 different ways.

This is only one example, and there are many other, where fields not only have different types between VS Code and language server, they also have different fields between the two.

Just want to check on, whether this is as designed, or is there an actual miss-alignment?

**Comment by r3m0t** [comment]
This is normal, there's two classes which convert from one to the other -  eg - https://github.com/microsoft/vscode-languageserver-node/blob/2041784436fed53f4e77267a49396bca22a7aacf/client/src/common/codeConverter.ts#L139

If you are writing an LSP be sure to import types from `vscode-languageserver-types` and not from `vscode.d.ts`

If you're writing middleware you will work with the VSCode types.

**Comment by dbaeumer** [comment]
You are correct. The mismatch is on purpose since LSP can be implement by language different than TypeScript.
